### ISSUE TITLE
Agenda item and attachment link extraction from pdf text

### ISSUE THREAD

**Issue by swotai** [codeforsanjose/city-agenda-scraper#35]
Based on the output of #34, extract corresponding agenda items and attachment links.

This work is very similar to what [Roland has presented previously](https://github.com/shengxio/aws-comp-nlp). If we get permission to use their code that'll be great. Otherwise, we can also do it.

- [ ] Extract agenda items using regex. This involves potentially mapping out all the relevant generalized agenda items
- [ ] Define the data structure of the output. This is also very similar to what Roland has demoed on July 29th
![image](https://user-images.githubusercontent.com/9345420/127757010-b18da14e-2e7f-4187-9a0c-2c34b919418e.png)

**Comment by xconnieex** [comment]
It's not clear to me how they are utilizing the subID and mainID columns. I think that's something else we need to look into and try to understand. mainID seems like headings within the agenda, which currently the scraping team doesn't grab since they only get the items that have an attachment file in Legistar. However the concept of subID is similar to what we discussed and could work, but we may need an additional level of subID to account for items with multiple attachments.

![image](https://user-images.githubusercontent.com/10645901/127759988-beada5de-9211-4c73-a2f6-a319d3d2f5c6.png)

Our version:
![image](https://user-images.githubusercontent.com/10645901/127760051-39df1079-96a2-44e5-8008-ac3f1a0b75d0.png)

The version I mocked up seems comprehensive but also seems like it stores a lot of redundant data due to the potential for multiple attachments. Maybe there's a different way to do it? Roland's way looks simpler because all agenda items are associated with only the one PDF as far as I can see.

If the scraping team can provide all the relevant PDFs we can try to use Roland's method and "match up" what we can find using regex and what the scraping team gets from the attachments so we have both sources. Otherwise we could try to get to the attachments ourselves through the PDFs.
